### ISSUE TITLE
How to Not Display "Off-Hours" in plot

### ISSUE THREAD

**Issue by dbii** [leeoniya/uPlot#261]
I have data that only comes in during business hours, 8-5. Is there a way to not show the times with no data (e.g. flat lines) (same with weekends, instead of showing 2 days of flat lines, just skip them).

My demo of the overnight issue (I can add weekend if needed for a different solution): https://jsfiddle.net/o05ftz4g/2/

<img width="905" alt="Screen Shot 2020-06-18 at 3 53 09 PM" src="https://user-images.githubusercontent.com/26703/85070696-e340f400-b17b-11ea-8235-faf6d4a8da66.png">

**Comment by dbii** [comment]
Another trick is to also have the x-axis ticks show up, just at 8:00 instead of midnight.

**Comment by leeoniya** [comment]
> Is there a way to not show the times with no data (e.g. flat lines) (same with weekends, instead of showing 2 days of flat lines, just skip them).

yes, there is:

```js
scales: {
  x: {
    distr: 2
  }
},
```

this creates equal distribution between every x datapoint. but it comes with a price: you lose "nice" temporal breakpoints like start of month, or start of day because what happens is that axis splits are simply plucked from every 2nd, 5th, 10th, 20th datapoint. it effectively treats the x scale as indexed/numeric for the purposes axis splitting, but still applies the `time: true` formatters to the timestamps in the data at those splits.

this is the configuration used by the stock demo to avoid showing gaps during non-trading days (weekends): https://leeoniya.github.io/uPlot/demos/candlestick-ohlc.html

> Another trick is to also have the x-axis ticks show up, just at 8:00 instead of midnight.

once you do the above, it's on you to implement your own x axis splitter which i guess would scan your data and return the indices in your data with timestamps that represent the 8:00 in your data. you may have to insert those data markers in advance if you don't have data which lands exactly on 8:00. and that's the other limitation: everything on a `distr: 2` scale deals in data indices, which means you cannot zoom to half-way between index 6 and 7 - it will always snap to an exact index because interpolating between datapoints in an unpredicatable scale and/or sparse dataset is not a simple matter.

**Comment by dbii** [comment]
I think I have it. I calculated and manually added the splits (as the indices of the x-Data times), and I was able to keep the original way to calculate values. I got rid of incrs since it didn't seem like that was relevant anymore.

https://jsfiddle.net/o05ftz4g/5/

This solution accommodates missing data values too (e.g. what is 8:00 doesn't exist), it is all based on indices. I have in the real data generator a means of filling in early data and missing data points, so that won't be an issue for me.
